src/view/com/composer/Composer.tsx

```diff
@@ -741,6 +750,19 @@ function useAnimatedBorders() {
   }
 }
 
+function useKeyboardVerticalOffset() {
+  const {height: windowHeight} = useWindowDimensions()
+
+  const [viewHeight, setViewHeight] = useState(0)
+
+  return [
+    isIOS ? windowHeight - viewHeight : 0,
```

Comment:
`onLayout` is async so `viewHeight` will be `0` on first render. How are we handling this? Why doesn't this result in flicker?

Reply:
Good question - I think it's hidden by the animation, but we can guard against this potential problem by defaulting to `windowHeight - 69`, meaning the value is 69 on first render (same as if we'd hard coded it)

```js
const {height: windowHeight} = useWindowDimensions()
// estimate height to avoid potential jumping on first render
const [viewHeight, setViewHeight] = useState(windowHeight - 69)
```

Reply:
<img width="708" alt="Screenshot 2024-06-13 at 01 05 10" src="https://github.com/bluesky-social/social-app/assets/10959775/f3552233-9edb-4744-a99f-3968b459dc99">

Reply:
(it's really dumb, I know)

Reply:
Sorry I keep asking silly questions but I want to understand:

- Does this height depend on any UI elements in the app (e.g. our modal styling) or is it platform-specific?
- Is this height the same across all devices, or does it depend on device?
- On a single device, is it consistent for this screen or does it depend on something?
- If it doesn't depend on the actual UI and is specific to the device, is there any way to measure it beforehand so that by this point it's always available?

Reply:
1. it's platform specific (specifically this native sheet presentation)
2. depends on device. the iPhone SE has no notch, so this area is small (40px). the notched iPhones are ~60px. the newest iPhones with the pill-shaped notch have the largest area of 69px
3. I'd expect it to stay the same. maybe rotating an iPad will change it?
4. I found this issue: https://github.com/react-navigation/react-navigation/issues/10317
Someone claims you can just add 10px to the top safe area inset to get the right value, so I'm gonna see if that works

Reply:
Update: I was able to simplify it to this:

```js
function useKeyboardVerticalOffset() {
  const {top} = useSafeAreaInsets()

  // Android etc
  if (!isIOS) return 0

  // iPhone SE
  if (top === 20) return 40

  // all other iPhones
  return top + 10
}
```

Weirdly the iPhone SE needs an extra 10px, but overall huge improvement.

Thanks for calling out my overly complex approach!